components/script/bluetooth_blacklist.rs

```diff
@@ -0,0 +1,98 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use regex::Regex;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Read;
+use std::path::Path;
+use std::string::String;
+
+const EXCLUDE_READS: &'static str = "exclude-reads";
+const EXCLUDE_WRITES: &'static str = "exclude-writes";
+const FILE_PATH: &'static str = "./resources/gatt_blacklist.txt";
+const VALID_UUID_REGEX: &'static str = "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}";
+
+thread_local!(pub static BLUETOOTH_BLACKLIST: RefCell<Option<BluetoothBlacklist>> =
+              RefCell::new(BluetoothBlacklist::init()));
+
+pub struct BluetoothBlacklist(HashMap<String, ValueType>);
```

Comment:
Because of traits we should create a HashMap data member for the struct(so we have to differ from the spec), which implements it, or return with a HashMap if we call fill_blacklist(). Currently there is one global instance, which is accessible from every file, on the other hand if we use the trait, there would be at least 5 of the same HashMap.